ui/src/components/DiffTextArea/index.jsx

```diff
@@ -0,0 +1,203 @@
+    theme.palette.type === 'light'
+      ? fade(theme.palette.common.black, 0.23)
+      : fade(theme.palette.common.white, 0.23);
+
+  return {
+    '@global': {
+      '.d2h-wrapper': {
+        '& .d2h-file-wrapper': {
+          border: 'none',
+        },
+      },
+      '.d2h-file-header': {
+        display: 'none',
+      },
+      '.d2h-info': {
+        display: 'none',
+      },
+      '.d2h-diff-table': {
+        '& .d2h-cntx': {
+          background: theme.palette.background.paper,
+        },
+        '& .d2h-code-side-linenumber.d2h-cntx': {
+          background: theme.palette.background.paper,
+          color: theme.palette.text.secondary,
+        },
+        '& .d2h-code-side-linenumber.d2h-del': {
+          color: THEME.PRIMARY_TEXT_DARK,
+        },
+        '& .d2h-code-side-linenumber.d2h-ins': {
+          color: THEME.PRIMARY_TEXT_DARK,
+        },
+        '& .d2h-code-side-linenumber': {
+          border: `solid ${theme.palette.divider}`,
+          borderWidth: `0 1px 0 1px`,
+          cursor: 'text',
+        },
+        '& .d2h-code-side-linenumber.d2h-code-side-emptyplaceholder.d2h-cntx.d2h-emptyplaceholder': {
+          background: theme.palette.grey['500'],
+        },
+        '& .d2h-ins': {
+          borderColor: 'none',
+          backgroundColor: theme.palette.diff.green.line,
+          color: THEME.PRIMARY_TEXT_DARK,
+        },
+        '& .d2h-del': {
+          borderColor: 'none',
+          backgroundColor: theme.palette.diff.red.line,
+          color: THEME.PRIMARY_TEXT_DARK,
+        },
+        '& .d2h-code-line ins, & .d2h-code-side-line ins': {
+          backgroundColor: theme.palette.diff.green.word,
+        },
+        '& .d2h-code-line del, & .d2h-code-side-line del': {
+          backgroundColor: theme.palette.diff.red.word,
+        },
+        '& .d2h-code-side-emptyplaceholder, & .d2h-emptyplaceholder': {
+          backgroundColor: theme.palette.grey['500'],
+        },
+      },
+    },
+    tab: {
+      flexGrow: 1,
+    },
+    tabContent: {
+      marginTop: theme.spacing(1),
+      padding: theme.spacing(1),
+    },
+    diffContainer: {
+      borderColor,
+      borderWidth: 1,
+      borderRadius: theme.shape.borderRadius,
+      borderStyle: 'solid',
+    },
+  };
+});
+
+/**
+ * An input text field with a diff view feature.
+ * Refer to `mozilla-frontend-infra/components` MarkdownTextArea components
+ * ref: https://github.com/mozilla-frontend-infra/components/blob/master/src/components/MarkdownTextArea/index.jsx
+ */
+function DiffTextArea(props) {
+  const { classes, onChange, rows, initialValue, ...rest } = props;
+  const [tabIndex, setTabIndex] = useState(props.defaultTabIndex);
+  const [value, setValue] = useState(props.value);
+  const isViewDiff = tabIndex === 1;
+  const isControlled =
+    'value' in props && props.value !== undefined && props.value !== null;
+  const isSorted =
+    'sort' in props && props.sort !== undefined && props.sort !== null;
```

Comment:
`sort` should be by design. No need to add a `sort` prop.

Reply:
OK, I see. Thanks for the review! I'll update it ASAP

Reply:
Just curious will we use `<DiffTextArea>` in other displays in the future?

Reply:
I think so. We will probably be using `DiffTextArea` in different views of the UI.

Reply:
Hmm, if we use it in different views. Do we also need to `sort` the text? For example, if we want to see the update of a `description`. It should still display diff if we change the line's order.

Reply:
Ahhh I see what you mean. You're absolutely right. I guess it makes sense to keep the `sort` prop!

Reply:
OK, thanks for the reply! If there is anything I should modify. Let me know!